ducktape/cluster/vagrant.py

```diff
@@ -22,12 +26,47 @@ class VagrantCluster(JsonCluster):
     """
     An implementation of Cluster that uses a set of VMs created by Vagrant. Because we need hostnames that can be
     advertised, this assumes that the Vagrant VM's name is a routeable hostname on all the hosts.
+
+    - If cluster_file is specified in the constructor's kwargs (i.e. passed via command line argument --cluster-file)
+      - If cluster_file exists on the filesystem, read cluster info from the file
+      - Otherwise, retrieve cluster info via "vagrant ssh-config" from vagrant and write cluster info to cluster_file
+    - Otherwise, retrieve cluster info via "vagrant ssh-config" from vagrant
     """
 
-    def __init__(self):
-        hostname, ssh_hostname, username, flags = None, None, None, ""
-        nodes = []
+    def __init__(self, *args, **kwargs):
+        self.is_aws = self._is_aws()
```

Comment:
This line results in opening a subprocess that calls `vagrant status` regardless of whether cluster info is cached, which results in unnecessary overhead.

I think we can simply make `self.is_aws` a property that gets lazily evaluated. E.g.

```
    @property
    def is_aws(self):
        if self._is_aws is None:
            proc = subprocess.Popen("vagrant status", shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
            output, _ = proc.communicate()
            self._is_aws = output.find("aws") >= 0

        return self._is_aws
```